MLIR.cmake

```diff
@@ -52,300 +55,59 @@ message(STATUS "LLVM_PROJECT_LIB        : " ${LLVM_PROJECT_LIB})
-    if (CMAKE_BUILD_TYPE)
-      set(LLVM_PROJ_BIN ${LLVM_PROJ_BUILD}/${CMAKE_BUILD_TYPE}/bin)
-    else()
-      set(LLVM_PROJ_BIN ${LLVM_PROJ_BUILD}/Release/bin)
-    endif()
-  else()
-    set(LLVM_PROJ_BIN ${LLVM_PROJ_BUILD}/bin)
-  endif()
+  set(LLVM_PROJ_BIN ${LLVM_TOOLS_BINARY_DIR})
 endif()
 message(STATUS "LLVM_PROJ_BIN           : " ${LLVM_PROJ_BIN})
 
-# Include paths for MLIR
-if (USE_INSTALLED_LLVM)
-  set(LLVM_SRC_INCLUDE_PATH ${LLVM_PROJ_SRC}/include)
-else()
-  set(LLVM_SRC_INCLUDE_PATH ${LLVM_PROJ_SRC}/llvm/include)
-endif()
-set(LLVM_BIN_INCLUDE_PATH ${LLVM_PROJ_BUILD}/include)
-set(MLIR_SRC_INCLUDE_PATH ${LLVM_PROJ_SRC}/mlir/include)
-set(MLIR_BIN_INCLUDE_PATH ${LLVM_PROJ_BUILD}/tools/mlir/include)
-set(MLIR_TOOLS_DIR ${LLVM_PROJ_BIN})
+# FIXME: Remove after migration
+set(MLIR_TOOLS_DIR ${LLVM_TOOLS_BINARY_DIR})
 
 # ONNX-MLIR tools folder
-if(MSVC)
-  if (CMAKE_BUILD_TYPE)
-    set(ONNX_MLIR_TOOLS_DIR ${CMAKE_BINARY_DIR}/bin/${CMAKE_BUILD_TYPE})
-  else()
-    set(ONNX_MLIR_TOOLS_DIR ${CMAKE_BINARY_DIR}/bin/Release)
-  endif()
-else()
-  set(ONNX_MLIR_TOOLS_DIR ${CMAKE_BINARY_DIR}/bin)
-endif()
+set(ONNX_MLIR_TOOLS_DIR ${CMAKE_BINARY_DIR}/${CMAKE_CFG_INTDIR}/bin)
 message(STATUS "ONNX_MLIR_TOOLS_DIR     : " ${ONNX_MLIR_TOOLS_DIR})
+# Set output library path
+set(ONNX_MLIR_LIB_DIR ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/${CMAKE_CFG_INTDIR})
+
 set(ONNX_MLIR_LIT_TEST_SRC_DIR ${ONNX_MLIR_SRC_ROOT}/test/mlir)
 set(ONNX_MLIR_LIT_TEST_BUILD_DIR ${CMAKE_BINARY_DIR}/test/mlir)
 
-set(
-        MLIR_INCLUDE_PATHS
-        ${LLVM_SRC_INCLUDE_PATH};${LLVM_BIN_INCLUDE_PATH};${MLIR_SRC_INCLUDE_PATH};${MLIR_BIN_INCLUDE_PATH}
-)
-include_directories(${MLIR_INCLUDE_PATHS})
+include_directories(${LLVM_INCLUDE_DIRS};${MLIR_INCLUDE_DIRS})
 
-if (NOT USE_INSTALLED_LLVM)
-  # Force CMAKE_INSTALL_PREFIX and BUILD_SHARED_LIBS to be the same as LLVM build
+# If we're pointing at an LLVM build, then assume we want to install into the same
+# install location
+if(EXISTS ${LLVM_PROJ_BUILD}/CMakeCache.txt)
   file(STRINGS ${LLVM_PROJ_BUILD}/CMakeCache.txt prefix REGEX CMAKE_INSTALL_PREFIX)
   string(REGEX REPLACE "CMAKE_INSTALL_PREFIX:PATH=" "" prefix ${prefix})
+  string(REGEX REPLACE "//.*" "" prefix ${prefix})
   set(CMAKE_INSTALL_PREFIX ${prefix} CACHE PATH "" FORCE)
   message(STATUS "CMAKE_INSTALL_PREFIX    : " ${CMAKE_INSTALL_PREFIX})
-
-  file(STRINGS ${LLVM_PROJ_BUILD}/CMakeCache.txt shared REGEX BUILD_SHARED_LIBS)
-  string(REGEX REPLACE "BUILD_SHARED_LIBS:BOOL=" "" shared ${shared})
-  set(BUILD_SHARED_LIBS ${shared} CACHE BOOL "" FORCE)
-  message(STATUS "BUILD_SHARED_LIBS       : " ${BUILD_SHARED_LIBS})
 endif()
 
+# Force BUILD_SHARED_LIBS to be the same as LLVM build
+set(shared ${LLVM_ENABLE_SHARED_LIBS})
+set(BUILD_SHARED_LIBS ${shared} CACHE BOOL "" FORCE)
+message(STATUS "BUILD_SHARED_LIBS       : " ${BUILD_SHARED_LIBS})
+
 # Threading libraries required due to parallel pass execution.
+# FIXME Should be added to LLVMConfig.cmake
```

Comment:
I am not sure this is needed - each of the libs coming from LLVM/MLIR carry with them their dependencies, so there's no need to explicitly find these packages. Am I missing something?